test/GraphQL.Conventions.Tests/Adapters/Engine/DependencyInjectionTests.cs

```diff
@@ -199,39 +198,31 @@ private class ScopedDocumentExecuter : GraphQL.DocumentExecuter
         {
             protected override IExecutionStrategy SelectExecutionStrategy(ExecutionContext context)
             {
-                var injector = GetChildInjector(context.GetDependencyInjector());
+                var injector = GetChildInjector(context.RequestServices);
                 return new ScopedExecutionStrategy(injector, base.SelectExecutionStrategy(context));
             }
 
-            private IDependencyInjector GetChildInjector(IDependencyInjector dependencyInjector)
-                => dependencyInjector.Resolve<ChildDependencyInjector>() ?? dependencyInjector;
+            private IServiceProvider GetChildInjector(IServiceProvider serviceProvider)
+                => serviceProvider.GetService<ChildDependencyInjector>() ?? serviceProvider;
 
             private class ScopedExecutionStrategy : IExecutionStrategy
             {
-                private readonly IDependencyInjector _injector;
+                private readonly IServiceProvider _serviceProvider;
                 private readonly IExecutionStrategy _innerStrategy;
 
-                public ScopedExecutionStrategy(IDependencyInjector injector, IExecutionStrategy innerStrategy)
+                public ScopedExecutionStrategy(IServiceProvider serviceProvider, IExecutionStrategy innerStrategy)
                 {
-                    _injector = injector;
+                    _serviceProvider = serviceProvider;
                     _innerStrategy = innerStrategy;
                 }
 
 
-                public async Task<ExecutionResult> ExecuteAsync(ExecutionContext context)
+                public Task<ExecutionResult> ExecuteAsync(ExecutionContext context)
                 {
-                    var key = typeof(IDependencyInjector).FullName ?? nameof(IDependencyInjector);
-                    var outerInjector = context.UserContext[key];
-
-                    try
-                    {
-                        context.UserContext[key] = _injector;
-                        return await _innerStrategy.ExecuteAsync(context);
-                    }
-                    finally
+                    return _innerStrategy.ExecuteAsync(new ExecutionContext(context)
```

Comment:
Note that `ExecutionContext` does not fully clone here, as noted in the xml comments of the constructor.  However, this is simply a test, and the test passes, so I did not change it further.